model/src/main/java/org/projectnessie/api/v2/http/HttpTreeApi.java

```diff
@@ -0,0 +1,723 @@
+    @APIResponse(responseCode = "404", description = "Ref doesn't exists"),
+    @APIResponse(responseCode = "409", description = "update conflict"),
+  })
+  SingleReferenceResponse deleteReference(
+      @Parameter(
+              description = "Optional expected type of the reference being deleted",
+              examples = {@ExampleObject(ref = "referenceType")})
+          @QueryParam("type")
+          Reference.ReferenceType type,
+      @Parameter(
+              schema = @Schema(pattern = REF_NAME_PATH_ELEMENT_REGEX),
+              description = CHECKED_REF_DESCRIPTION,
+              examples = {
+                @ExampleObject(ref = "ref"),
+                @ExampleObject(ref = "refWithHash"),
+                @ExampleObject(ref = "refDefault"),
+              })
+          @PathParam("ref")
+          String ref)
+      throws NessieConflictException, NessieNotFoundException;
+
+  @Override
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{ref:" + REF_NAME_PATH_ELEMENT_REGEX + "}/contents/{key}")
+  @Operation(
+      summary = "Get the content object associated with a key.",
+      description =
+          "This operation returns the content value for a content key at a particular point in history as defined "
+              + "by the 'ref' parameter.",
+      operationId = "getContentV2")
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Information for a table, view or another content object for the given key",
+        content =
+            @Content(
+                mediaType = MediaType.APPLICATION_JSON,
+                examples = {@ExampleObject(ref = "iceberg")},
+                schema = @Schema(implementation = org.projectnessie.model.Content.class))),
+    @APIResponse(responseCode = "400", description = "Invalid input, ref name not valid"),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(
+        responseCode = "403",
+        description = "Not allowed to view the given reference or read object content for a key"),
+    @APIResponse(
+        responseCode = "404",
+        description = "Table not found on 'ref' or non-existent reference")
+  })
+  ContentResponse getContent(
+      @Parameter(description = KEY_PARAMETER_DESCRIPTION) @PathParam("key") ContentKey key,
+      @Parameter(
+              schema = @Schema(pattern = REF_NAME_PATH_ELEMENT_REGEX),
+              description = REF_PARAMETER_DESCRIPTION,
+              examples = {
+                @ExampleObject(ref = "ref"),
+                @ExampleObject(ref = "refWithHash"),
+                @ExampleObject(ref = "refDefault"),
+                @ExampleObject(ref = "refDetached"),
+              })
+          @PathParam("ref")
+          String ref)
+      throws NessieNotFoundException;
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{ref:" + REF_NAME_PATH_ELEMENT_REGEX + "}/contents")
```

Comment:
The new storage stuff could return contents using paging, and have start+end key constraints. Can you support for paging for `getSeveralContents` + `getMultipleContents`, too?`

Reply:
Certainly. That is not directly related to the `ref` param, though, is it?

Reply:
#5591